.github/workflows/sonarcloud.yml

```
@@ -1,26 +1,54 @@
+# This GitHub action will checkout and scan third party code.
+# Please ensure that any changes to this action do not perform
+# actions that may result in code from that branch being executed
+# such as installing dependencies or running build scripts.
+
 name: SonarCloud
 
 on:
-  workflow_call:
-    secrets:
-      SONAR_TOKEN:
-        required: true
+  workflow_run:
+    workflows:
+      - Run tests
+    types:
+      - completed
+
+permissions:
+  actions: read
 
 jobs:
   sonarcloud:
+    # Only scan code from non-forked repositories that have passed the tests
+    # This will skip scanning the code for forks, but will run for the main repository on PRs from forks
+    if: ${{ github.event.workflow_run.conclusion == 'success' && github.event.workflow_run.repository.fork == false }}
     name: SonarCloud
     runs-on: ubuntu-latest
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4
         with:
+          repository: ${{ github.event.workflow_run.head_repository.full_name }} # Use the repository that triggered the workflow
+          ref: ${{ github.event.workflow_run.head_branch }} # Use the branch that triggered the workflow
           fetch-depth: 0  # Shallow clones should be disabled for better relevancy of analysis
 
       - name: Download artifacts
         uses: actions/download-artifact@v4
         with:
           name: lcov.info
           path: coverage
+          github-token: ${{ github.token }} # This is required when downloading artifacts from a different repository or from a different workflow run.
+          run-id: ${{ github.event.workflow_run.id }} # Use the workflow id that triggered the workflow
+
+      - name: Download sonar-project.properties
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          REPOSITORY: metamask/metamask-extension
```

Nit: It doesn't really matter since the URL is not case-sensitive, but our organization name is `MetaMask`

```suggestion
          REPOSITORY: MetaMask/metamask-extension
```